pydantic/types.py

```diff
@@ -127,6 +145,45 @@ def conbytes(*, strip_whitespace: bool = False, min_length: int = None, max_leng
     return type('ConstrainedBytesValue', (ConstrainedBytes,), namespace)
 
 
+T = TypeVar("T")
+
+
+class ConstrainedList(List[T]):
+    __origin__ = list  # Needed for pydantic to detect that this is a list
+
+    min_items: Optional[int] = None
+    max_items: Optional[int] = None
+    item_type: Type[T]
+
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        yield cls.list_length_validator
+
+    @classmethod
+    def list_length_validator(cls, v: 'List[T]', field: 'Field', config: 'BaseConfig') -> 'List[T]':
+        v_len = len(v)
+
+        min_items = cls.min_items
+        if min_items is not None and v_len < min_items:
+            raise errors.ListMinLengthError(limit_value=min_items)
+
+        max_items = cls.max_items
+        if max_items is not None and v_len > max_items:
+            raise errors.ListMaxLengthError(limit_value=max_items)
+
+        return v
+
+
+def conlist(item_type: Type[T], *, min_items: int = None, max_items: int = None) -> Type[List[T]]:
+    # use kwargs then define conf in a dict to aid with IDE type hinting
```

Comment:
not sure this comment makes sense here.

Reply:
Me neither, but copied from `constr()`

Reply:
Maybe it makes more sense there. Probably remove.

Reply:
Done